lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java

```diff
@@ -200,20 +200,38 @@ public APIGatewayProxyResponseEvent handleRequest(
                     HttpStatusCode.FORBIDDEN,
                     new CommonExpressOAuthError(
                             OAuth2Error.ACCESS_DENIED, customOAuth2ErrorDescription));
+        } catch (NumberFormatException e) {
+            LOGGER.error(String.format("Parsing value of %s to int failed", MAXIMUM_ATTEMPT_COUNT));
+            eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatusCode.INTERNAL_SERVER_ERROR,
+                    new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
         } catch (OAuthHttpResponseExceptionWithErrorBody e) {
             eventProbe.counterMetric(LAMBDA_CHECK_PASSPORT_COMPLETED_ERROR);
 
-            LOGGER.debug(e.getMessage(), e);
+            // Debug only as Oauth errors appear in the redirect url
+            // This will output the specific error message
+            if (LOGGER.isDebugEnabled()) {
+                String customOAuth2ErrorDescription = e.getErrorReason();
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        e.getStatusCode(), // Status Code determined by throw location
+                        new CommonExpressOAuthError(
+                                OAuth2Error.SERVER_ERROR, customOAuth2ErrorDescription));
+            }
 
+            // Non-debug route - standard OAuth2Error.SERVER_ERROR
             return ApiGatewayResponseGenerator.proxyJsonResponse(
                     e.getStatusCode(), // Status Code determined by throw location
                     new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
-        } catch (Exception e) {
+        } catch (RuntimeException e) {
```

Comment:
This should still be Exception. We shouldnt ever be in a position where there is a checked exception were not catching elsewhere but I always prefer to play it safe.

Reply:
Reverted back to Exception